src/runtimes/index.js

```diff
@@ -1,8 +1,82 @@
+const { extname, basename } = require('path')
+
 const goRuntime = require('./go')
 const jsRuntime = require('./node')
 const rustRuntime = require('./rust')
 
+/**
+ * Finds functions for a list of paths using a specific runtime. The return
+ * value is an object containing an array of the functions found (`functions`)
+ * and an array with the paths that haven't been recognized by the runtime
+ * (`remainingPaths`).
+ *
+ * @param   {Array<String>} paths
+ * @param   {Object} runtime
+ * @returns {Promise<Object>}
+ */
+const findFunctionsInRuntime = async function ({ dedupe = false, paths, runtime }) {
+  const functions = await runtime.findFunctionsInPaths(paths)
+
+  // If `dedupe` is true, we use the function name (`filename`) as the map key,
+  // so that `function-1.js` will overwrite `function-1.go`. Otherwise, we use
+  // `srcPath`, so that both functions are returned.
+  const key = dedupe ? 'name' : 'srcPath'
+
+  // Augmenting the function objects with additional information.
+  const augmentedFunctions = functions.map((func) => [
```

Comment:
These parameters could easily be returned by each runtime, but for convenience we compute them upstream so that runtimes don't have to return them, for convenience. They are generic parameters that can be inferred from the caller.